### ISSUE TITLE
[Feature Request] IndexOIDFormat for indirect TAG query

### ISSUE THREAD

**Issue by pavleec** [toni-moreno/snmpcollector#234]
Hello,

I'm trying to add Cisco CPU utilization with the card names send as tags, based on the following:

1.snmpwalk for the OID (**1.3.6.1.4.1.9.9.109.1.1.1.1.2** ) for the object "cpmCPUTotalPhysicalIndex" gives me the PhysicalIndex mapping of cards like below:

```
SNMPv2-SMI::enterprises.9.9.109.1.1.1.1.2.2 = INTEGER: 2359704
SNMPv2-SMI::enterprises.9.9.109.1.1.1.1.2.18 = INTEGER: 10154515
SNMPv2-SMI::enterprises.9.9.109.1.1.1.1.2.34 = INTEGER: 33511382

```

2.It is albo possible to figure out which card is what by polling OID (**1.3.6.1.2.1.47.1.1.1.1.7**) for object "entPhysicalName" using the values ($VAL1) received in step 1

```
SNMPv2-SMI::mib-2.47.1.1.1.1.7.2359704 = STRING: "0/0/* - cpu"
SNMPv2-SMI::mib-2.47.1.1.1.1.7.10154515 = STRING: "0/1/* - cpu"
SNMPv2-SMI::mib-2.47.1.1.1.1.7.33511382 = STRING: "0/2/* - cpu"

```
3.Now it's time to snmpwalk for the OID (**1.3.6.1.4.1.9.9.109.1.1.1.1.7**) for the object "cpmCPUTotal1minRev" which gives the CPU utilization percent for one minute for the (which is $IDX1) from step 1.
```

SNMPv2-SMI::enterprises.9.9.109.1.1.1.1.7.2 = Gauge32: 2
SNMPv2-SMI::enterprises.9.9.109.1.1.1.1.7.18 = Gauge32: 1
SNMPv2-SMI::enterprises.9.9.109.1.1.1.1.7.34 = Gauge32: 1
```

I was not able to achive it right in snmpcollector so I guess it would be great feature to add an option to "Index win indirect TAG" mode  to define an IndexOIDFormat the same like is used for IndexTAGFormat. This way I could use $IDX1 instead of $VAL1 for indexing. What do you think about implementing this?

In short indexing looks like below (marked in bold):
SNMPv2-SMI::enterprises.9.9.109.1.1.1.1.2.**2** = INTEGER: **2359704**
SNMPv2-SMI::mib-2.47.1.1.1.1.7.**2359704** = STRING: "0/0/* - cpu"
SNMPv2-SMI::enterprises.9.9.109.1.1.1.1.7.**2** = Gauge32: 2

**Comment by toni-moreno** [comment]
While  cpmCPUTotalPhysicalIndex values will be stored as $VAL1 ,  entPhysicalName will be stored as $VAL2 so you will want to config your indexTagFormat as "$VAL1 - $VAL2" if you want.

**Comment by pavleec** [comment]
That would work partially and properly format only the tag label. To get the cpmCPUTotal1minRev values $IDX1 should be used instead of $VAL1.

**Comment by sbengo** [comment]
Hi @pavleec

I don't really underestand your issue. When you define a measurement you have 3 different modes:
- Value (`value`) : makes a `get` on SNMPMetrics
- Direct (`indexed`): makes a `walk` and stores values: $IDX1, $VAL1
- Indirect (`indexed_it`): makes 2 walks, 1 to get the real `Index OID` - `$IDX1, $VAL1` and another one: `TagOID` - `$IDX2, $VAL2`

On your case, you have to use the `indexed_it` in order to:
- First, get the indexes for the metric used on the measurement. As you said, the output of this first walk will return the following:
````
WALK - 1.3.6.1.4.1.9.9.109.1.1.1.1.2 :
SNMPv2-SMI::enterprises.9.9.109.1.1.1.1.2.2 = INTEGER: 2359704
SNMPv2-SMI::enterprises.9.9.109.1.1.1.1.2.18 = INTEGER: 10154515
SNMPv2-SMI::enterprises.9.9.109.1.1.1.1.2.34 = INTEGER: 33511382
````
On this case, the SNMPCollector will store the vars:
````
$IDX1 = [2,18,34]
$VAL1 = [2359704, 10154515, 33511382]
````

- Second, as the measurement is defined as `Indirect` it will do a second walk over the `Tag OID`, in this case, defined as the entPhysicalName:

````
WALK - 1.3.6.1.2.1.47.1.1.1.1.7 
SNMPv2-SMI::mib-2.47.1.1.1.1.7.2359704 = STRING: "0/0/* - cpu"
SNMPv2-SMI::mib-2.47.1.1.1.1.7.10154515 = STRING: "0/1/* - cpu"
SNMPv2-SMI::mib-2.47.1.1.1.1.7.33511382 = STRING: "0/2/* - cpu"
````
And the SNMPCollector will store the vars:

````
$IDX2 = [2359704, 10154515, 33511382]
$VAL2 = [0/0/* - cpu, 0/1/* - cpu, 0/2/* - cpu]
````
Note that the $VAL2 and $IDX1 are the same value

- Third, define your `Tag Format`

Following with your example the:
`$IDX1, $VAL2` will grant you the output:
````
2 - 0/0/* - cpu
18 - 0/1/* - cpu
34 - 0/0/* - cpu
````

Is this your desired behaivour?

**Comment by pavleec** [comment]
Yes, that's exactly what I wanted to achive :) Thanks a lot for a detailed explanation. I guess the above example describes perfectly differences in polling methods and should be added to the snmpcollector Wiki manual. I just wasn't sure that $IDX1 and $VAL1 variables are also available in the indirect  mode. As requested feature already exists please clase this issue.

**Comment by toni-moreno** [comment]
ok @pavleec

**Comment by sbengo** [comment]
Actually, we discovered those indirect index cases when we were working with the same device as yours!

On the wiki, there is an example to do a basic configuration of a Cisco Catalyst switch, within an explanantion of the indirect index to get cpu values: [Basic configuration - Cisco Catalyst Switch](https://github.com/toni-moreno/snmpcollector/wiki/Example:-Configuring-Cisco-Catalyst-Switch)

Greeting!
